doc/sphinx/user-extensions/syntax-extensions.rst

```diff
@@ -1573,6 +1538,8 @@ Numeral notations
 
    .. exn:: Syntax error: [prim:reference] expected after 'Notation' (in [vernac:command]).
 
+     .. todo PR at least a couple of these are repeated below.  What do you suggest?
```

Comment:
I suggest removing the first occurrence and adjusting the description of the second one to include both `Numeral Notation` and `String Notation`.

Reply:
There were 6 in common, they're now belong String Notation.  Two of them are syntax errors--we usually don't document those, and they might show up for other commands.  Take a look.

Reply:
I think the syntax errors are important to document, at least currently.  Maybe once the manual is rewritten that will no longer be the case, but currently syntax errors are rather impenetrable, I think.  Furthermore, these are things that you might reasonably expect the system to handle, and it doesn't.  So, for example, if you write `Definition (id nat) := nat.`, this seems like a reasonable thing to be a syntax error of "I expected a name after definition, you gave me an open parenthesis".  But if you write `String Notation (list char) (@id _) (@id _) : list_char_scope`, the appropriate error message is "String Notation doesn't yet support types which are not single qualified identifiers" not "expected qualid, got open paren".  Or, worse, `Numeral Notation Z (@id _) (@id _) : Z_scope` should give "the parsing and printing functions must be a single qualified identifier" not "expected qualid after qualid, got open paren".

Reply:
We can keep these 2, but it seems like the number of possible errors is vast, so unlikely to add more now.  We should do better with the error messages so they can be understood without having the read the manual beyond the grammar itself.  That likely requires significant work on the parser.

Reply:
@JasonGross Another solution would be to actually parse terms but reject anything that is not a qualid at interpretation time, with a clear error message.

Reply:
> @JasonGross Another solution would be to actually parse terms but reject anything that is not a qualid at interpretation time, with a clear error message.

Perhaps we should do that